### ISSUE TITLE
extends(xrpToDrops: invalid value 'NaN', should be a number matching (^-?[0-9]*\.?[0-9]*$).))]

### ISSUE THREAD

**Issue by cujo30227** [ripple/ripple-lib#1399]
I *sometimes* get the error when trying to send a payment. 

extends(xrpToDrops: invalid value 'NaN', should be a number matching (^-?[0-9]*\.?[0-9]*$).))]

After waiting for a while, and reconnecting, it usually works. Anybody has any idea what could be the reason and how to fix?

The Payment JSON and instructions are defined in JS as:

```
var instructions = {
                                   maxLedgerVersionOffset: 30
                               };

var payment = {
                             source: {
                                   address: fromacc,
                                   maxAmount: {
                                          value: xrpAmount + "",
                                          currency: 'XRP',
                                   },
                                   tag: parseInt(sourceTag + "")
                              },
                              destination: {
                                   address: destination,
                                   amount: {
                                          value: xrpAmount + "",
                                          currency: 'XRP',
                              },
                              tag: parseInt(destTag + "")
                              }
                        };
                      
```

**Comment by yxxyun** [comment]
Did you connect to the s2.ripple.com?

**Comment by cujo30227** [comment]
Hi,
I did initially, and realized that this was consistently leading to the error. Now I'm connecting to s1.ripple.com (or wss://s-west.ripple.com, wss://s-east.ripple.com), and still sometimes receive the error..

**Comment by intelliot** [comment]
I suspect this is related to https://github.com/ripple/ripple-lib/issues/1398 (caused by rippled reporting mode; may be fixed in the next release of rippled)

**Comment by OliverRedmanCooke** [comment]
I am also experiencing this error. @intelliot Do you have any estimations for when this will be fixed?

**Comment by intelliot** [comment]
PRs are welcome 😇

**Comment by intelliot** [comment]
Note that s1.ripple.com is not intended for use in production. You may consider running your own rippled server, or connecting to xrplcluster.com.

**Comment by tacyarg** [comment]
> Did you connect to the s2.ripple.com?

I was, once I switched to the "xrplcluster" as @intelliot suggested the issue went away.

**Comment by cujo30227** [comment]
Hi, I've also started to connect to xrplcluster , and the problem went away..
Thanks

**Comment by OliverRedmanCooke** [comment]
Thanks everyone, I will try the cluster instead.

**Comment by intelliot** [comment]
This was fixed in ripple-lib v1.9.5

(PR #1401)

Please open a new issue if you see this reoccur.
